### PR TITLE
Update mintWithSignature ABIs with proper return types

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
The focus of this PR is to update return types and method signatures related to `mintWithSignature` across ERC721, ERC20, and ERC1155 contracts.

### Detailed summary
- Updated return types for `mintWithSignature`
- Adjusted method signatures for `mintWithSignature`
- Refactored `simulateTransaction` test
- Updated contract ABIs for ERC721, ERC20, and ERC1155

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/scripts/generate/abis/erc1155/ISignatureMintERC1155.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [
   "event TokensMintedWithSignature(address indexed signer, address indexed mintedTo, uint256 indexed tokenIdMinted, (address to, address royaltyRecipient, uint256 royaltyBps, address primarySaleRecipient, uint256 tokenId, string uri, uint256 quantity, uint256 pricePerToken, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) mintRequest)",
-  "function mintWithSignature((address to, address royaltyRecipient, uint256 royaltyBps, address primarySaleRecipient, uint256 tokenId, string uri, uint256 quantity, uint256 pricePerToken, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) payload, bytes signature) payable returns (address signer)",
+  "function mintWithSignature((address to, address royaltyRecipient, uint256 royaltyBps, address primarySaleRecipient, uint256 tokenId, string uri, uint256 quantity, uint256 pricePerToken, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) payload, bytes signature) payable",
```
Comment (jnsdls) [inline packages/thirdweb/scripts/generate/abis/erc1155/ISignatureMintERC1155.json:3]: no return type at all?

File: packages/thirdweb/scripts/generate/abis/erc20/ISignatureMintERC20.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [
   "event TokensMintedWithSignature(address indexed signer, address indexed mintedTo, (address to, address primarySaleRecipient, uint256 quantity, uint256 price, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) mintRequest)",
-  "function mintWithSignature((address to, address primarySaleRecipient, uint256 quantity, uint256 price, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) payload, bytes signature) payable returns (address signer)",
+  "function mintWithSignature((address to, address primarySaleRecipient, uint256 quantity, uint256 price, address currency, uint128 validityStartTimestamp, uint128 validityEndTimestamp, bytes32 uid) payload, bytes signature) payable",
```
Comment (jnsdls) [inline packages/thirdweb/scripts/generate/abis/erc20/ISignatureMintERC20.json:3]: ditto?

File: packages/thirdweb/src/transaction/actions/simulate.test.ts [hunk 3]
```diff
@@ -17,5 +16,19 @@ it.runIf(process.env.TW_SECRET_KEY)(
       from: TEST_WALLET_A,
     });
     expect(result).toMatchInlineSnapshot("[]");
-  },
-);
+  });
+
+  test("should return human-readable", async () => {
```
Comment (jnsdls) [inline packages/thirdweb/src/transaction/actions/simulate.test.ts:21]: good test to add thx!

**Review by jnsdls — APPROVED** [review]
looks good, I assume the erc1155 and erc20 fn signatures simply *are* empty returns?

File: packages/thirdweb/scripts/generate/abis/erc1155/ISignatureMintERC1155.json (hunk 1 above)
Comment (gregfromstl) [reply packages/thirdweb/scripts/generate/abis/erc1155/ISignatureMintERC1155.json:3]: Yeah the erc1155 and erc20 ones don't return anything: https://github.com/thirdweb-dev/contracts/blob/main/contracts/prebuilts/token/TokenERC1155.sol

**Pull request merged**